### PR TITLE
fix(serve): do not set port for modern client

### PR DESCRIPTION
**Pull request by snitin315**
**What kind of change does this PR introduce?**
fix

**Did you add tests for your changes?**
NA

**If relevant, did you update the documentation?**
NA
**Summary**

we removed `client.port` in https://github.com/webpack/webpack-dev-server/pull/3181

Refer https://github.com/webpack/webpack-dev-server/pull/3181#issuecomment-818921391

**Does this PR introduce a breaking change?**
No

**Other information**
No

### CODE REVIEW

File: packages/serve/src/startDevServer.ts [hunk 1]
```diff
@@ -72,9 +72,17 @@ export default async function startDevServer(
         const options = mergeOptions(compilerWithDevServerOption.options.devServer || {}, devServerCliOptions);
 
         if (isDevServer4) {
+            // eslint-disable-next-line node/no-extraneous-require
+            const devServerFlags = require('webpack-dev-server/bin/cli-flags').devServer;
+
+            const isModernClient = Boolean(devServerFlags.find((flag) => flag.name === 'client-web-socket-url'));
+
             options.port = await findPort(options.port);
             options.client = options.client || {};
```
Comment (alexander-akait) [inline packages/serve/src/startDevServer.ts:81]: We move it under `isModerClient` too

Comment (snitin315) [reply packages/serve/src/startDevServer.ts:81]: Oh yes!

**Comment by alexander-akait** [comment]
To be honestly I don't understand why we do it, if client port is not specified or `0` we just get `self.location.port` (basic workflow), so when you run without proxy, we already get port from browser, if we proxy hostname developer uses the same port so `self.location.port` will work, only one situation possible when you proxy dev server on other port, but in this case you need manually set `client.port`, because we don't know original hostname/port

**Comment by alexander-akait** [comment]
It was my mistake here, we should not do it here, we should improve it on dev server, i.e. `options.client.port || options.port`, when we do release and improve `public` here https://github.com/webpack/webpack-dev-server/pull/3181/ and do release again

Difficult...

**Pull request merged**